libraries/AC_PrecLand/AC_PrecLand.cpp

```diff
@@ -251,3 +188,172 @@ void AC_PrecLand::handle_msg(mavlink_message_t* msg)
+                    } else {
+                        float NIS_x = _ekf_x.getPosNIS(_target_pos_rel_meas_NED.x, xy_pos_var);
+                        float NIS_y = _ekf_y.getPosNIS(_target_pos_rel_meas_NED.y, xy_pos_var);
+                        if (MAX(NIS_x, NIS_y) < 3.0f || _outlier_reject_count >= 3) {
+                            _outlier_reject_count = 0;
+                            _ekf_x.fusePos(_target_pos_rel_meas_NED.x, xy_pos_var);
+                            _ekf_y.fusePos(_target_pos_rel_meas_NED.y, xy_pos_var);
+                            _last_update_ms = AP_HAL::millis();
+                            _target_acquired = true;
+                        } else {
+                            _outlier_reject_count++;
+                        }
+                    }
+                }
+            }
+
+            // Output prediction
+            if (target_acquired()) {
+                _target_pos_rel_est_NE.x = _ekf_x.getPos();
+                _target_pos_rel_est_NE.y = _ekf_y.getPos();
+                _target_vel_rel_est_NE.x = _ekf_x.getVel();
+                _target_vel_rel_est_NE.y = _ekf_y.getVel();
+
+                run_output_prediction();
+            }
+            break;
+        }
+    }
+}
+
+bool AC_PrecLand::retrieve_los_meas(Vector3f& target_vec_unit_body)
+{
+    if (_backend->have_los_meas() && _backend->los_meas_time_ms() != _last_backend_los_meas_ms) {
+        _last_backend_los_meas_ms = _backend->los_meas_time_ms();
+        _backend->get_los_body(target_vec_unit_body);
+
+        // Apply sensor yaw alignment rotation
+        float sin_yaw_align = sinf(radians(_yaw_align*0.01f));
+        float cos_yaw_align = cosf(radians(_yaw_align*0.01f));
+        Matrix3f Rz = Matrix3f(
+            cos_yaw_align, -sin_yaw_align, 0,
+            sin_yaw_align, cos_yaw_align, 0,
+            0, 0, 1
+        );
+
+        target_vec_unit_body = Rz*target_vec_unit_body;
+        return true;
+    } else {
+        return false;
+    }
+}
+
+void AC_PrecLand::run_output_prediction()
+{
+    _target_pos_rel_out_NE = _target_pos_rel_est_NE;
+    _target_vel_rel_out_NE = _target_vel_rel_est_NE;
+
+    // Predict forward from delayed time horizon
+    for (uint8_t i=0; i<_inertial_history.size(); i++) {
+        const struct inertial_data_frame_s& inertial_data = _inertial_history.peek(i);
+        _target_vel_rel_out_NE.x -= inertial_data.correctedVehicleDeltaVelocityNED.x;
+        _target_vel_rel_out_NE.y -= inertial_data.correctedVehicleDeltaVelocityNED.y;
+        _target_pos_rel_out_NE.x += _target_vel_rel_out_NE.x * inertial_data.dt;
+        _target_pos_rel_out_NE.y += _target_vel_rel_out_NE.y * inertial_data.dt;
+    }
+
+    // compensate for camera offset from the center of the vehicle
+    Vector3f land_ofs_ned_m = _ahrs.get_rotation_body_to_ned() * Vector3f(_land_ofs_cm_x,_land_ofs_cm_y,0) * 0.01f;
+    _target_pos_rel_est_NE.x += land_ofs_ned_m.x;
+    _target_pos_rel_est_NE.y += land_ofs_ned_m.y;
```

Comment:
Yes, you're right about the _target_pos_rel_*out*_NE.  Fixed in the follow-up commit, "fix bug with land offset"